signer/core/signed_data.go

```diff
@@ -728,8 +762,11 @@ func formatPrimitiveValue(encType string, encValue interface{}) string {
 	if strings.HasPrefix(encType, "bytes") {
 		return fmt.Sprintf("%s", encValue)
 	} else if strings.HasPrefix(encType, "uint") || strings.HasPrefix(encType, "int") {
-		bigIntValue, _ := encValue.(*big.Int)
-		return fmt.Sprintf("%d (0x%x)", bigIntValue, bigIntValue)
+		b, err := parseIntegerType(encType, encValue)
+		if err != nil {
+			return fmt.Sprintf("ERROR: %v", err)
```

Comment:
is there a reason why you don't want to return an `error` here?

Reply:
Yes, laziness. Ostensibly, we should never reach here, since things are validated earlier. However, you're right that we should actually return an error even so, for robustness sake